### PR TITLE
Trainmultiqc

### PR DESCRIPTION
**Pull request by U13bs1125**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/pairgenomealign/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/pairgenomealign _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/last/train/main.nf [hunk 1]
```diff
@@ -14,6 +14,7 @@ process LAST_TRAIN {
     output:
     tuple val(meta), path("*.train"), emit: param_file
     path "versions.yml"           , emit: versions
```
Comment (charles-plessy) [inline modules/nf-core/last/train/main.nf:16]: Please make the `versions` channel the last of the output.

File: modules/nf-core/last/train/main.nf [hunk 2]
```diff
@@ -31,6 +32,28 @@ process LAST_TRAIN {
...
+    echo "# section_name: 'Alignment parameters and summary'" >> ${prefix}_mqc.tsv
+    echo "# format: 'tsv'" >> ${prefix}_mqc.tsv
+    echo "# plot_type: 'table'" >> ${prefix}_mqc.tsv
+    echo "# description: 'This plot shows the last alignment parameters'" >> ${prefix}_mqc.tsv
+    echo "# pconfig:" >> ${prefix}_mqc.tsv
+    echo "#    id: 'alingment parameters'" >> ${prefix}_mqc.tsv
+    echo "#    title: 'alingment parameters'" >> ${prefix}_mqc.tsv
+    echo "#    ylab: ''" >> ${prefix}_mqc.tsv
+    echo "id\tsubstitution_percent_identity\tlast -t\tlast -a\tlast -A\tlast -b\tlast -B\tlast -S" >> ${prefix}_mqc.tsv
+    for i in *.\$INDEX_NAME.train
```
Comment (charles-plessy) [inline modules/nf-core/last/train/main.nf:45]: There will only be a single `.train` file in the work directory of the `last/train` module, so you do not need a `for` loop.

File: subworkflows/local/pairalign_m2m/main.nf [hunk 3]
```diff
@@ -109,6 +109,8 @@ workflow PAIRALIGN_M2M {
 
     emit:
 
+    train = LAST_TRAIN.out.param_file
```
Comment (charles-plessy) [inline subworkflows/local/pairalign_m2m/main.nf:112]: If the `mqc.tsv` files are produced by the `last/train` module, we do not need to export the parameter files.

File: subworkflows/local/pairalign_m2m/main.nf [hunk 4]
```diff
@@ -109,6 +109,8 @@ workflow PAIRALIGN_M2M {
...
+    tsv = LAST_TRAIN.out.tsv
```
Comment (charles-plessy) [inline subworkflows/local/pairalign_m2m/main.nf:113]: Please rename the channel `multiqc`; I think that other workflows or modules use that name too.

File: modules/local/custommoduletrain.nf
Comment (charles-plessy) [inline modules/local/custommoduletrain.nf:1]: If the `last/train` module does all the work, we do not need this file anymore.

Comment (U13bs1125) [reply modules/local/custommoduletrain.nf:1]: ok

File: subworkflows/local/pairalign_m2m/main.nf (hunk 4 above)
Comment (U13bs1125) [reply subworkflows/local/pairalign_m2m/main.nf]: done

File: subworkflows/local/pairalign_m2m/main.nf (hunk 3 above)
Comment (U13bs1125) [reply subworkflows/local/pairalign_m2m/main.nf]: yes

File: modules/nf-core/last/train/main.nf (hunk 1 above)
Comment (U13bs1125) [reply modules/nf-core/last/train/main.nf]: alright

**Comment by charles-plessy** [comment]
Thanks to the `last/train` update we can collect directly training parameters without using the custom module, like in #7

**Comment by charles-plessy** [comment]
Closing because we added the custommodule in a different way.

**Pull request closed without merging**